### PR TITLE
bsc: add support for bsc as it doesnt have 1559

### PR DESCRIPTION
**Pull request by emailtovamos**
Addresses https://github.com/stackup-wallet/stackup-bundler/issues/125#issuecomment-1488351995

### CODE REVIEW

File: pkg/modules/checks/standalone.go [hunk 1]
```diff
@@ -61,7 +66,9 @@ func (s *Standalone) ValidateOpValues() modules.UserOpHandlerFunc {
 		g.Go(func() error { return ValidateVerificationGas(ctx.UserOp, s.maxVerificationGas) })
 		g.Go(func() error { return ValidatePaymasterAndData(ctx.UserOp, gc, gs) })
 		g.Go(func() error { return ValidateCallGasLimit(ctx.UserOp) })
-		g.Go(func() error { return ValidateFeePerGas(ctx.UserOp, gbf) })
+		if !isBSC(ctx.ChainID) {
+			g.Go(func() error { return ValidateFeePerGas(ctx.UserOp, gbf) })
+		}
```
Comment (hazim-j) [inline pkg/modules/checks/standalone.go:71]: I don't think switching off this sanity check based on hardcoded chain ID values is the best approach for this. Instead we should have a fallback mechanism if chains don't support EIP-1559 style gas fees.

**Review by hazim-j — CHANGES_REQUESTED** [review]
Thanks for the PR @emailtovamos. Left my comment below.

File: pkg/modules/checks/standalone.go (hunk 1 above)
Comment (emailtovamos) [reply pkg/modules/checks/standalone.go:71]: Ok, how could we know programmatically if chains don't support  EIP-1559 style gas fees apart from knowing their chain IDs?
And this fallback mechanism should be part of `ValidateFeePerGas()` function or somewhere else?

**Comment by weiihann** [comment]
@emailtovamos did it work on your end by adding BSC chainID check? I tried running this bundler version with a BSC testnet node, but it doesn't work.

still investigating this issue, most likely work needs to be done on the handleOps. @hazim-j it'll be great if we can get some pointers so that we can contribute to this feature.

**Comment by emailtovamos** [comment]
@weiihann could you describe what exactly you tried doing which didn't work?

**Comment by emailtovamos** [comment]
@weiihann Did it fail while sending user operation? In that case what code did you use? 

Or did it fail while simply running the bundler itself? Because this works for me.

**Comment by weiihann** [comment]
> @weiihann Did it fail while sending user operation? In that case what code did you use?
> 
> Or did it fail while simply running the bundler itself? Because this works for me.

It failed when I sent the user operation. I used the codes in https://github.com/stackup-wallet/erc-4337-examples

**Pull request closed without merging**